packages/nimbus/src/components/number-input/number-input.slots.tsx

```diff
@@ -55,7 +62,7 @@ export const NumberInputRootSlot = withProvider<
 export const NumberInputInputSlot = withContext<
   HTMLInputElement,
   NumberInputInputSlotProps
->("input", "input");
+>(RaInput, "input");
```

Comment:
Having a context-aware react-aria input here means it will handle displaying all the fancy `data-` attrs we've come to know and love (focus, invalid, etc)